framework/src/utils/MooseUtils.C

```diff
@@ -809,6 +810,20 @@ template std::string join<std::vector<MooseEnumItem>>(const std::vector<MooseEnu
 template std::string join<std::set<MooseEnumItem>>(const std::set<MooseEnumItem> &,
                                                    const std::string &);
 
+void
+createSymlink(const std::string & target, const std::string & link)
+{
+  clearSymlink(link);
+  symlink(target.c_str(), link.c_str());
+}
+
+void
+clearSymlink(const std::string & link)
+{
+  struct stat sbuf;
+  if (lstat(link.c_str(), &sbuf) == 0)
+    unlink(link.c_str());
```

Comment:
Check return codes of these low level Unix APIs

Reply:
Just noting that we haven't used `unlink` before, or for awhile. We've been using `std::remove` from `<cstdio>`. I _do_ think this is the right API as the C library version isn't guaranteed to work for non-regular files (like symlinks). Eventually this will get replaced when we go to the C++17 standard.